front-end/src/pages/ActivityModal.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
```

Comment:
Activities seems like a better name for this component to me.  mostly it is a list that contains a modal.